pkg/run/apply_manifests_test.go

```diff
@@ -0,0 +1,92 @@
+package run_test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/fluxcd/flux2/pkg/manifestgen/install"
+	"github.com/fluxcd/pkg/ssa"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/weave-gitops/pkg/logger/loggerfakes"
+	"github.com/weaveworks/weave-gitops/pkg/run"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// mock ssa.ResourceManager
+type mockResourceManagerForApply struct {
+	run.ResourceManagerForApply
+	state stateApply
+}
+
+type stateApply string
+
+const (
+	stateApplyAllReturnErr   stateApply = "apply-all-return-err"
+	stateWaitForSetReturnErr stateApply = "wait-for-set-return-err"
+	applyAllErrorMsg                    = "apply all error"
+	waitForSetErrorMsg                  = "wait for set error"
+)
+
+func (man *mockResourceManagerForApply) ApplyAll(ctx context.Context, objects []*unstructured.Unstructured, opts ssa.ApplyOptions) (*ssa.ChangeSet, error) {
+	switch man.state {
+	case stateApplyAllReturnErr:
+		return nil, errors.New(applyAllErrorMsg)
+
+	default:
+		changeSet := ssa.NewChangeSet()
+
+		return changeSet, nil
+	}
+}
+
+func (man *mockResourceManagerForApply) WaitForSet(set object.ObjMetadataSet, opts ssa.WaitOptions) error {
+	switch man.state {
+	case stateWaitForSetReturnErr:
+		return errors.New(waitForSetErrorMsg)
+
+	default:
+		return nil
+	}
+}
+
+var _ = Describe("Apply", func() {
+	var fakeLogger *loggerfakes.FakeLogger
+	var fakeContext context.Context
+	var fakeInstallOptions install.Options
+	var fakeManifestsContent []byte
+
+	BeforeEach(func() {
+		fakeLogger = &loggerfakes.FakeLogger{}
+		fakeContext = context.Background()
+		fakeInstallOptions = install.MakeDefaultOptions()
+
+		fakeManifests, err := install.Generate(fakeInstallOptions, "")
+		Expect(err).NotTo(HaveOccurred())
+		fakeManifestsContent = []byte(fakeManifests.Content)
+	})
+
+	It("should apply manifests successfully", func() {
+		man := &mockResourceManagerForApply{}
+
+		_, err := run.Apply(fakeLogger, fakeContext, man, fakeManifestsContent)
```

Comment:
We're totally inconsistent on this, and I think we use this pattern more often than not - but we don't have to make `Apply` (and several other examples) public just to be able to test it, we could just call the test package `run` instead of `run_test`, and then the tests have access to private functions, too.

Reply:
Changed all instances of `package run_test` to `package_run`.

Made only required functions public.